### ISSUE TITLE
testing:  Test minio server with restic project and do load testing. 

### ISSUE THREAD

**Issue by harshavardhana** [minio/minio#1024]

**Comment by JaCoB1123** [comment]
I just tested restic with minio and got the following error
[log.txt](https://github.com/minio/minio/files/97335/log.txt)

**Comment by harshavardhana** [comment]
Thanks @JaCoB1123 . Looks like a windows issue taking a look.

**Comment by harshavardhana** [comment]
Thanks @JaCoB1123 this is a Golang runtime bug and perhaps a known issue will be fixed only in Go 1.6 release which will happen in couple of weeks relevant issue to be noted here. 

```
runtime: free list of span 0x26d7100:
0xc082580900 -> 0x5400410044 (BAD)
fatal error: free list corrupted

runtime stack:
runtime.throw(0x9cbe00, 0x13)
    c:/go/src/runtime/panic.go:527 +0x97
runtime.mSpan_Sweep(0x26d7100, 0x2100000000, 0xc000000001)
    c:/go/src/runtime/mgcsweep.go:186 +0x80e
runtime.sweepone(0x46e389)
    c:/go/src/runtime/mgcsweep.go:97 +0x162
runtime.gosweepone.func1()
    c:/go/src/runtime/mgcsweep.go:109 +0x28
runtime.systemstack(0xc08203a1c0)
    c:/go/src/runtime/asm_amd64.s:262 +0x7e
runtime.mstart()
    c:/go/src/runtime/proc1.go:674

goroutine 3 [running]:
runtime.systemstack_switch()
    c:/go/src/runtime/asm_amd64.s:216 fp=0xc082029f38 sp=0xc082029f30
runtime.gosweepone(0x1)
    c:/go/src/runtime/mgcsweep.go:110 +0x44 fp=0xc082029f60 sp=0xc082029f38
runtime.bgsweep(0xc082036000)
    c:/go/src/runtime/mgcsweep.go:54 +0xc4 fp=0xc082029f98 sp=0xc082029f60
runtime.goexit()
    c:/go/src/runtime/asm_amd64.s:1696 +0x1 fp=0xc082029fa0 sp=0xc082029f98
created by runtime.gcenable
    c:/go/src/runtime/mgc.go:206 +0x5a

goroutine 1 [select]:
github.com/minio/minio/vendor/github.com/minio/minio-xl/pkg/minhttp.ListenAndServeLimited(0x0, 0xc08202a270, 0x1, 0x1, 0x8)
    C:/Git/go/src/github.com/minio/minio/vendor/github.com/minio/minio-xl/pkg/minhttp/http_windows.go:205 +0x4bd
main.startServer(0xc08202e4e0, 0x5, 0x0, 0xc082002280, 0x8, 0xa, 0x0, 0x0, 0x0, 0x0, ...)
    C:/Git/go/src/github.com/minio/minio/server-main.go:151 +0x110
main.serverMain(0xc0820d60f0)
```

https://github.com/golang/go/issues/13287

**Comment by harshavardhana** [comment]
@JaCoB1123 can you paste "minio.exe --debug" output?

**Comment by JaCoB1123** [comment]
I get `Unknown flag. ‘--debug’`. Not sure if this helps you anyway.
Passing `--debug` to `minio server` does not work either.

```
C:\Git\go\bin>minio.exe --debug
Unknown flag. ‘--debug’

NAME:
  Minio - Cloud Storage Server for Micro Services.

DESCRIPTION:
  Micro services environment provisions one Minio server per application instance. Scalability is achieved through large number of smaller personalized instances. This version of the Minio binary is built using Filesystem storage backend for magnetic and solid state disks.

USAGE:
  minio [flags] command [arguments...]

COMMANDS:
  server        Start Minio cloud storage server.
  config        Collection of config management commands.
  version       Print version.
  update        Check for a new software update.

FLAGS:
  --config-folder, -C "C:\Users\Jan Bader\.minio"       Path to configuration folder.
  --address ":9000"                                     ADDRESS:PORT for cloud storage access.
  --cert                                                Provide your domain certificate.
  --key                                                 Provide your domain private key.
  --json                                                Enable json formatted output.

VERSION:
  UNOFFICIAL.GOGET

MEM:
  Used: 413kB | Allocated: 413kB | Used-Heap: 413kB | Allocated-Heap: 950kB

PLATFORM:
  Host: JBTHINKCENTRE | OS: windows | Arch: amd64

RUNTIME:
  Version: go1.5.1 | CPUs: 8
```

**Comment by JaCoB1123** [comment]
I just rebuilt with go1.6beta2 and now the backup completes without an issue.

**Comment by harshavardhana** [comment]
Thanks I will reproduce this problem, hopefully now we know that it's golang 1.5.1 that's the issue.

**Comment by harshavardhana** [comment]
Not able to reproduce this with golang1.5.1 , looks like a rare problem. @JaCoB1123 can you tell us more on how the issue is reproduced or basically are you able to see it all the time when you use restic with minio?  - thanks

**Comment by JaCoB1123** [comment]
Sorry, I didn't notice the notification until now.
The folder to backup contained multiple big files (>500MB, Database Backups). The error occured after a few seconds and not instantly. Maybe your dataset wasn't large enough?

**Comment by harshavardhana** [comment]
> Sorry, I didn't notice the notification until now.
> The folder to backup contained multiple big files (>500MB, Database Backups). The error occured after a few seconds and not instantly. Maybe your dataset wasn't large enough?

My data was around 100MB, if its possible would you mind giving it a try and reproducing again?

**Comment by JaCoB1123** [comment]
I didn't have Go 1.5 installed anymore so I reinstalled 1.5.3.
- updated restic to restic/restic@7a8054d6784105e60750824636ded46f524ff7c6
- updated minio to 53a983659e1094fd891730162c99f27e7b873509  
  There was a newer commit available but it wouldn't build because of `assertFS`
- rebuilt both
- Created a new Folder for minio
- started minio  
  `minio.exe server D:\minio >` [log.txt](https://github.com/minio/minio/files/119216/log.txt) `2>&1`
- inited new restic repository  
  `restic.exe -r s3:http://127.0.0.1:9000/restic init >` [log_restic.txt](https://github.com/minio/minio/files/119217/log_restic.txt) `2>&1`
- backed up Directory with a single MS SQL Server Backup File (426 MiB)
  `restic.exe -r s3:http://127.0.0.1:9000/restic backup D:\test >` [log_restic2.txt](https://github.com/minio/minio/files/119218/log_restic2.txt) `2>&1`

Results:
- restic echos `panic:` and freezes
- minio stops with error
- restic-Bucket is 307 MiB afterwards

**Comment by harshavardhana** [comment]
> I didn't have Go 1.5 installed anymore so I reinstalled 1.5.3.
> 
> updated restic to restic/restic@7a8054d
> updated minio to 53a9836
> There was a newer commit available but it wouldn't build because of assertFS
> rebuilt both
> Created a new Folder for minio
> started minio
> minio.exe server D:\minio > log.txt 2>&1
> inited new restic repository
> restic.exe -r s3:http://127.0.0.1:9000/restic init > log_restic.txt 2>&1
> backed up Directory with a single MS SQL Server Backup File (426 MiB) restic.exe -r s3:http://127.0.0.1:9000/restic backup D:\test > log_restic2.txt 2>&1
> Results:
> 
> restic echos panic: and freezes
> minio stops with error
> restic-Bucket is 307 MiB afterwards

Perfect thank you, this is what is needed - the bug is really easy to reproduce on your windows node. Can you tell me more about the system details like windows version, etc. ? So that i can try to replicate your setup - Thanks.

**Comment by JaCoB1123** [comment]
I'm running Windows 10 64-bit on a Machine with an i7-4770 and 16 GB RAM.
Anything else I can provide?

**Comment by harshavardhana** [comment]
> I'm running Windows 10 64-bit on a Machine with an i7-4770 and 16 GB RAM.
> Anything else I can provide?

Thanks - can you also paste 'go env' ?

**Comment by JaCoB1123** [comment]
I'm on another machine right now. Will post it on Monday.

**Comment by harshavardhana** [comment]
> I'm on another machine right now. Will post it on Monday.

Sure no problem thank you.

**Comment by JaCoB1123** [comment]
As promised, here's the output of `go env`:

```
set GOARCH=amd64
set GOBIN=
set GOEXE=.exe
set GOHOSTARCH=amd64
set GOHOSTOS=windows
set GOOS=windows
set GOPATH=C:\Git\go\;C:\Go
set GORACE=
set GOROOT=C:\Go
set GOTOOLDIR=C:\Go\pkg\tool\windows_amd64
set GO15VENDOREXPERIMENT=
set CC=gcc
set GOGCCFLAGS=-m64 -mthreads -fmessage-length=0
set CXX=g++
set CGO_ENABLED=1
```

**Comment by harshavardhana** [comment]
> set GOARCH=amd64
> set GOBIN=
> set GOEXE=.exe
> set GOHOSTARCH=amd64
> set GOHOSTOS=windows
> set GOOS=windows
> set GOPATH=C:\Git\go\;C:\Go
> set GORACE=
> set GOROOT=C:\Go
> set GOTOOLDIR=C:\Go\pkg\tool\windows_amd64
> set GO15VENDOREXPERIMENT=
> set CC=gcc
> set GOGCCFLAGS=-m64 -mthreads -fmessage-length=0
> set CXX=g++
> set CGO_ENABLED=1

Very cool, will try to reproduce. Thanks again.

**Comment by harshavardhana** [comment]
> Very cool, will try to reproduce. Thanks again.

We moved to default 1.6 with current master, this issue is not reproducible anymore. Setting #pending-release flag.

**Comment by harshavardhana** [comment]
New minio server released built with go 1.6, should fix this issue closing.
